app/helpers/affiliate_helper.rb

```diff
@@ -12,4 +12,8 @@ def render_embed_code_javascript(affiliate)
     JS
     javascript_tag(embed_code)
   end
+
+  def favicon_url(url)
+    url.presence || '/favicon_affiliate.ico'
+  end
```

Comment:
We can leave this for now, but it's a little weird for the `affiliate_helper` to include a method that doesn't involve an affiliate. In my original comment I was thinking we'd pass the affiliate to the method (`favicon_url(affiliate)`). Your method is an improvement, but a better spot for it would be `app/helpers/url_helper.rb`. But again, we can leave it as-is for now.